src/Plugin/Field/FieldWidget/VisualisationFileWidget.php

```diff
@@ -177,6 +177,7 @@ protected function getVisualisationPlugin(FieldItemListInterface $items, $delta,
       'style' => [
         'plugin_id' => $this->getElementOptions($items, $delta, $form, $form_state, 'visualisation_style'),
       ],
+      'entity' => $items->getEntity(),
```

Comment:
I am not sure this changes anything in your solution but please remember that this needs to be optional. Hopefully, in the not-too-distant future we will be able to create visualisations with data coming from Views.

Reply:
Hey @jorgegc I added this to provide more value to the `hook_dvf_xxxx_alter` it has no immediate use but provides a huge amount of flexibility if themes/modules want to alter based on other field values, especially handy when used with govCMS. 

A lot more than this would need refactoring if it was optional as this method relies on `FieldItemListInterface` which is specific to an entity field item.

That said, I agree it should be optional in the vis plugin which i'll ensure will have a graceful fallback.

Reply:
Updated `VisualisationStyleInterface` to indicate this could be null. Have left it in this file for above reasons :)